python_modules/dagster/dagster/_core/definitions/configurable.py

```diff
@@ -63,6 +67,74 @@ def apply_config_mapping(self, config: Any) -> EvaluateValueResult:
         )
 
 
+def copy_with_default(old_field: Field, new_config_value: Any) -> Field:
+    return Field(
+        config=old_field.config_type,
+        default_value=new_config_value,
+        is_required=False,
+        description=old_field.description,
+    )
+
+
+def _apply_defaults_to_schema_field(field: Field, additional_default_values: Any) -> Field:
+    # This work by validating the top-level config and then
+    # just setting it at that top-level field. Config fields
+    # can actually take nested values so we only need to set it
+    # at a single level
+
+    evr = validate_config(field.config_type, additional_default_values)
```

Comment:
another nice benefit of this is that we'll now see errors with configured at definition time rather than during runtime.